src/modules/fw_pos_control_l1/fw_pos_control_l1_main.cpp

```diff
@@ -1350,94 +1364,144 @@ FixedwingPositionControl::control_position(const math::Vector<2> &current_positi
 
 		} else if (pos_sp_triplet.current.type == position_setpoint_s::SETPOINT_TYPE_TAKEOFF) {
 
-			/* Perform launch detection */
-			if (launchDetector.launchDetectionEnabled() &&
-					launch_detection_state != LAUNCHDETECTION_RES_DETECTED_ENABLEMOTORS) {
-				/* Inform user that launchdetection is running */
-				static hrt_abstime last_sent = 0;
-				if(hrt_absolute_time() - last_sent > 4e6) {
-					mavlink_log_critical(_mavlink_fd, "Launchdetection running");
-					last_sent = hrt_absolute_time();
+			if (_parameters.do_runway_takeoff == true) {
+				if (!_takeoff_initialized) {
+					_hdg_hold_yaw = _att.yaw;
```

Comment:
I think we should lock onto a heading that points into the direction of the takeoff waypoint. I'm pretty sure I can't place the airframe onto the runway so it has the perfect heading to follow it ;)